index.ts

```diff
@@ -0,0 +1,104 @@
+const NativeClient = require('bindings')('hawk_tracer_client').HawkTracerClient;   // loads build/Release/hawk_tracer_client.node
+
+export interface HawkTracerClientOptions {
+    source: string;
+    map_files?: string;
+}
+
+// HT_Event
+interface BareEvent {
+    klass_id: number;
+    timestamp: number;
+    id: number;
+}
+
+// HT_EventKlassInfoEvent
+interface EventKlassInfoEvent extends BareEvent {
+    info_klass_id: number;
+    event_klass_name: string;
+    field_count: number;
+}
+
+function isEventKlassInfoEvent(event: BareEvent): event is EventKlassInfoEvent {
+    const e = event as EventKlassInfoEvent;
+    return e.info_klass_id !== undefined && e.event_klass_name !== undefined;
+}
+
+export interface Event extends BareEvent {
+    klass_name: string;
+}
+
+function isEvent(event: Event): event is Event {
+    return event.klass_name !== undefined;
+}
+
+// HT_CallstackBaseEvent | HT_CallstackIntEvent | HT_CallstackStringEvent
+export interface CallstackEvent extends Event {
```

Comment:
Could we have `CallstackBaseEvent` and both `CallstackIntEvent` and `CallstackStringEvent` inheriting from that? We could then have `CallstackEvent` defined as `CallstackStringEvent | CallstackIntEvent`. Any reasons we try to merge those three classes in a single one?

Reply:
Typescript is statically typed language. For statically typed languages, it's generally not recommended to include type information in the type name.
Since the user can check the type of `label` anyway, I don't think having 3 separate classes have any benefit.